### ISSUE TITLE
Broken install in directory containing /

### ISSUE THREAD

**Issue by jamesvclements** [lovell/sharp#3703]
**TLDR**

Sharp fails to install when the parent directory contains a /

```
fatal error: 'vips/vips8' file not found
```

Initially I thought this was an issue with [Payload](https://github.com/payloadcms/create-payload-app/issues/17). I organize my projects like `Projects / Client/{project-name}`. For now I renamed the folder to not use a slash, but I wanted to report this incase anyone else ends up down a similar rabbit hole. I found this solve pretty randomly, most of my time was spent diagnosing M1 architecture and xcode installs found in relate threads to the `vips/vips8` error.

## Possible install-time or require-time problem

- [x] I have read the [documentation relating to installation](https://sharp.pixelplumbing.com/install).
- [x] I have ensured that the architecture and platform of Node.js used for `npm install` is the same as the architecture and platform of Node.js used at runtime.

### Are you using the latest version of sharp?

- [x] I am using the latest version of `sharp` as reported by `npm view sharp dist-tags.latest`.

### What is the complete output of running `npm install --verbose --foreground-scripts sharp` in an empty directory?

<details>

```
npm http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 272ms
sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.14.2/libvips-8.14.2-darwin-arm64v8.tar.br
sharp: Integrity check passed for darwin-arm64v8OST 200 https://registry.npmjs.org/-/npm/v1/security/advi
sh: prebuild-install: command not foundp fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advi
gyp info it worked if it ends with okttp fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advi
gyp verb cli [
gyp verb cli   '/usr/local/bin/node',
gyp verb cli   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'rebuild'
gyp verb cli ]
gyp info using node-gyp@9.3.1
gyp info using node@18.16.1 | darwin | arm64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directoryfetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advi
gyp verb command configure []
gyp verb find Python Python is not set from command line or npm configuration
gyp verb find Python Python is not set from environment variable PYTHON
gyp verb find Python checking if "python3" can be used
gyp verb find Python - executing "python3" to get executable path
gyp verb find Python - executable path is "/usr/local/opt/python@3.11/bin/python3.11"npm/v1/security/advi
gyp verb find Python - executing "/usr/local/opt/python@3.11/bin/python3.11" to get version
gyp verb find Python - version is "3.11.4"etch POST 200 https://registry.npmjs.org/-/npm/v1/security/advi
gyp info find Python using Python version 3.11.4 found at "/usr/local/opt/python@3.11/bin/python3.11"
gyp verb get node dir no --target version specified, falling back to host node version: 18.16.1
gyp verb command install [ '18.16.1' ]
gyp verb install input version string "18.16.1"POST 200 https://registry.npmjs.org/-/npm/v1/security/advi
gyp verb install installing version: 18.16.1
gyp verb install --ensure was passed, so won't reinstall if already installed
gyp verb install version is already installed, need to check "installVersion"
gyp verb got "installVersion" 9
gyp verb needs "installVersion" 9
gyp verb install version is good
gyp verb get node dir target node version installed: 18.16.1
gyp verb build dir attempting to create "build" dir: /Users/james/Local/Directory : With Slash/empty/node_modules/sharp/build
gyp verb build dir "build" dir needed to be created? Yes
gyp verb python symlink creating symlink to "/usr/local/opt/python@3.11/bin/python3.11" at "/Users/james/Local/Directory : With Slash/empty/node_modules/sharp/build/node_gyp_bins/python3"
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /Users/james/Local/Directory : With Slash/empty/node_modules/sharp/build/config.gypi
gyp verb config.gypi checking for gypi file: /Users/james/Local/Directory : With Slash/empty/node_modules/sharp/config.gypi
gyp verb common.gypi checking for gypi file: /Users/james/Local/Directory : With Slash/empty/node_modules/sharp/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn /usr/local/opt/python@3.11/bin/python3.11
gyp info spawn args [
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/james/Local/Directory : With Slash/empty/node_modules/sharp/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/james/Library/Caches/node-gyp/18.16.1/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/james/Library/Caches/node-gyp/18.16.1',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/james/Library/Caches/node-gyp/18.16.1/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/james/Local/Directory : With Slash/empty/node_modules/sharp',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp verb command build []ify:sharp: http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advi
gyp verb build type Release
gyp verb architecture arm64
gyp verb node dev dir /Users/james/Library/Caches/node-gyp/18.16.1
gyp verb `which` succeeded for `make` /usr/bin/make
gyp verb bin symlinks adding symlinks (such as Python), at "/Users/james/Local/Directory : With Slash/empty/node_modules/sharp/build/node_gyp_bins", to PATH
gyp info spawn make
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
  cc -o Release/obj.target/nothing/../node-addon-api/nothing.o ../../node-addon-api/nothing.c '-DNODE_GYP_MODULE_NAME=nothing' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' -I/Users/james/Library/Caches/node-gyp/18.16.1/include/node -I/Users/james/Library/Caches/node-gyp/18.16.1/src -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/openssl/config -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/openssl/openssl/include -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/uv/include -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/zlib -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/v8/include  -O3 -gdwarf-2 -mmacosx-version-min=10.15 -arch arm64 -Wall -Wendif-labels -W -Wno-unused-parameter -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/nothing/../node-addon-api/nothing.o.d.raw   -c
  rm -f Release/nothing.a && ./gyp-mac-tool filter-libtool libtool  -static -o Release/nothing.a Release/obj.target/nothing/../node-addon-api/nothing.o
warning: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: archive library: Release/nothing.a the table of contents is empty (no object file members in the library define global symbols)
  touch Release/obj.target/libvips-cpp.stampch POST 200 https://registry.npmjs.org/-/npm/v1/security/advi
  c++ -o Release/obj.target/sharp-darwin-x64/src/common.o ../src/common.cc '-DNODE_GYP_MODULE_NAME=sharp-darwin-x64' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DNAPI_VERSION=7' '-DBUILDING_NODE_EXTENSION' -I/Users/james/Library/Caches/node-gyp/18.16.1/include/node -I/Users/james/Library/Caches/node-gyp/18.16.1/src -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/openssl/config -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/openssl/openssl/include -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/uv/include -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/zlib -I/Users/james/Library/Caches/node-gyp/18.16.1/deps/v8/include -I../../node-addon-api -I../vendor/8.14.2/darwin-x64/include -I../vendor/8.14.2/darwin-x64/include/glib-2.0 -I../vendor/8.14.2/darwin-x64/lib/glib-2.0/include  -O3 -gdwarf-2 -mmacosx-version-min=10.13 -arch arm64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fexceptions -Wall -Oz -MMD -MF ./Release/.deps/Release/obj.target/sharp-darwin-x64/src/common.o.d.raw   -c
../src/common.cc:13:10: fatal error: 'vips/vips8' file not foundregistry.npmjs.org/-/npm/v1/security/advi
#include <vips/vips8>
         ^~~~~~~~~~~~
1 error generated.
make: *** [Release/obj.target/sharp-darwin-x64/src/common.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:203:23)
gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
gyp ERR! System Darwin 22.2.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/james/Local/Directory : With Slash/empty/node_modules/sharp
gyp ERR! node -v v18.16.1
gyp ERR! node-gyp -v v9.3.1
gyp ERR! not ok 
npm info run sharp@0.32.1 install { code: 1, signal: null }
npm verb stack Error: command failed
npm verb stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:53:27)
npm verb stack     at ChildProcess.emit (node:events:513:28)
npm verb stack     at maybeClose (node:internal/child_process:1091:16)
npm verb stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm verb pkgid sharp@0.32.1
npm verb cwd /Users/james/Local/Directory : With Slash/empty
npm verb Darwin 22.2.0
npm verb node v18.16.1
npm verb npm  v9.5.1
npm ERR! code 1
npm ERR! path /Users/james/Local/Directory : With Slash/empty/node_modules/sharp
npm ERR! command failed
npm ERR! command sh -c (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
npm verb exit 1
npm verb unfinished npm timer reify 1687300892119
npm verb unfinished npm timer reify:build 1687300893166
npm verb unfinished npm timer build 1687300893166
npm verb unfinished npm timer build:deps 1687300893166
npm verb unfinished npm timer build:run:install 1687300893168
npm verb unfinished npm timer build:run:install:node_modules/sharp 1687300893168
npm verb code 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/james/.npm/_logs/2023-06-20T22_41_32_097Z-debug-0.log
```

</details>

### What is the output of running `npx envinfo --binaries --system --npmPackages=sharp --npmGlobalPackages=sharp`?

```
  System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 5.43 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.16.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.5.1 - /usr/local/bin/npm
```

**Comment by lovell** [comment]
> prebuild-install: command not found

It looks like the `prebuild-install` dependency was not installed correctly. Have you tried the classic approach of deleting `node_modules` and trying again?

**Comment by jamesvclements** [comment]
> > prebuild-install: command not found
> 
> It looks like the `prebuild-install` dependency was not installed correctly. Have you tried the classic approach of deleting `node_modules` and trying again?

I did! I ended up reinstalling node, npm, restoring my shell to the default zsh (I was previously using fish which seemed to force my terminal into Intel-mode despite being on an M1 Mac), deleting my global npm cache, etc. 

This issue surface using the ‘npx create-payload-app’ command for Payload CMS, which installs sharp behind the scenes. That’s where the install was breaking. 

I tried reinstalling Xcode tools and manually installing libvips via Homebrew as well but the only thing that ended up working was moving out of my directory that contained a / in the name. For both a blank mode project just installing sharp and the create-payload command.
